brion/circuit.cpp

```diff
@@ -153,22 +174,52 @@ class Circuit::Impl
         return values;
     }
 
-    size_t getNumNeurons() const
+    virtual Vector3fs getPositions( const GIDSet& gids ) const{
+
+        const NeuronMatrix& data = get(
+            gids, brion::NEURON_POSITION_X | brion::NEURON_POSITION_Y |
+                 brion::NEURON_POSITION_Z );
+
+        GIDSet::const_iterator gid = gids.begin();
+        Vector3fs positions( gids.size( ));
+        #pragma omp parallel for
+        for( size_t i = 0; i < gids.size(); ++i )
+        {
+            try
+            {
+                positions[i] =
+                    brion::Vector3f( boost::lexical_cast< float >( data[i][0] ),
+                                     boost::lexical_cast< float >( data[i][1] ),
+                                     boost::lexical_cast< float >( data[i][2] ));
+            }
+            catch( const boost::bad_lexical_cast& )
+            {
+                LBWARN << "Error parsing circuit position or orientation for gid "
+                       << *gid << ". Morphology not transformed." << std::endl;
+                positions[i] = Vector3f::ZERO;
+            }
+            #pragma omp critical (brain_circuit_getPositions)
+            ++gid;
+        }
+        return positions;
+
+    }
+
+    virtual size_t getNumNeurons() const
     {
         return _table.find( SECTION_NEURONS )->second.size();
     }
 
-    Strings getTypes( const NeuronClass type ) const
+    virtual Strings getTypes( const NeuronClass type ) const
     {
         switch( type )
         {
-        case NEURONCLASS_ETYPE:
-        {
+        case NEURONCLASS_ETYPE:{
             CircuitTable::const_iterator i = _table.find( SECTION_ETYPES );
             return i != _table.end() ? i->second : Strings();
         }
 
-        case NEURONCLASS_MTYPE:
+        case NEURONCLASS_MTYPE:{
```

Comment:
?

Also, can you try to keep with the coding style of the respective project in the future?